### PR TITLE
Create codeql.yml

### PR DESCRIPTION
**Pull request by kanwalpreetd**
Added Github workflow for running security code scans through [CodeQL](https://docs.github.com/en/code-security/code-scanning/introduction-to-code-scanning/about-code-scanning).

The scan results will be visible under the [Security](https://github.com/stellar/stellar-core/security/code-scanning) tab, once a scan is complete.

### CODE REVIEW

**Comment by Shaptic** [comment]
This repository already has CodeQL scanning, please see the workflow checks, e.g. [this run](https://github.com/stellar/js-xdr/actions/runs/9036026304/job/24832018689?pr=123) on this PR's latest commit.

**Comment by Shaptic** [comment]
OH LOL THIS **_IS_** THE CODEQL CHECK :laughing: 

I was so confused why it wasn't happening on another PR

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+  schedule:
+    - cron: '35 22 * * 5'
```
Comment (Shaptic) [inline .github/workflows/codeql.yml:9]:
```suggestion
on:
  push:
    branches: [ "master" ]
  pull_request:
```

there's no reason to `cron` it if we're doing it on all pushes and pulls imo

**Review by Shaptic — APPROVED** [review]
just one note but otherwise LGTM!

File: .github/workflows/codeql.yml (hunk 1 above)
Comment (kanwalpreetd) [reply .github/workflows/codeql.yml:9]: hmm, I think maybe we keep the `cron` (incase if there is no activity for some time, the scan can still run and maybe find new issues if there are updates in the rules the scanner runs). `cron` will just run once a week

Comment (Shaptic) [reply .github/workflows/codeql.yml:9]: Ah good point!! Sounds good then

**Pull request merged**